### PR TITLE
Support self-hosted prompts in js sdk

### PR DESCRIPTION
**Pull request by yolken-airplane**
## Description
This change updates the js sdk to support self-hosted prompts.

## Testing
Testing completed successfully via deploying locally and verifying that prompts worked correctly in conjunction with https://github.com/airplanedev/web/pull/3458 and https://github.com/airplanedev/airport/pull/5771.

### CODE REVIEW

File: packages/airplane/internal/api/client.ts [hunk 1]
```diff
@@ -506,6 +517,65 @@ export class Client {
...
 
+  async createPromptSelfHostedOutputs(
+    prompt: CreatePromptRequest,
+    opts: { abortSignal?: AbortSignal; idempotencyKey?: string } = {},
+  ): Promise<string> {
+    const agentHost = this.agentStorageInternalHost;
+    const runID = this.agentStorageRunID;
+
+    const agentPath = "/v0/dp/prompts/create";
+    const apiPath = "/v0/prompts/create";
```
Comment (ameyakhare) [inline packages/airplane/internal/api/client.ts:528]: why these vars? afaict they're only used in 1 place

File: packages/airplane/internal/api/client.ts [hunk 2]
```diff
@@ -733,10 +803,37 @@ export class Client {
...
+    // Get the prompt from the API
+    const apiResp = await this.fetcher.get<{
       prompt: Prompt;
     }>("/v0/prompts/get", { id }, { abortSignal: opts.abortSignal });
-    return resp.prompt;
+
+    if (apiResp.prompt.zoneToken) {
+      // There's a zone token set, so use that to get the real schema and values
+      const agentHost = this.agentStorageInternalHost;
+      const runID = this.agentStorageRunID;
```
Comment (ameyakhare) [inline packages/airplane/internal/api/client.ts:814]: ditto

**Review by ameyakhare — APPROVED** [review]

File: packages/airplane/internal/api/client.ts (hunk 1 above)
Comment (yolken-airplane) [reply packages/airplane/internal/api/client.ts:528]: No idea, I was just copying this pattern from elsewhere in the SDK. :-)

File: packages/airplane/internal/api/client.ts (hunk 2 above)
Comment (yolken-airplane) [reply packages/airplane/internal/api/client.ts]: Removed these.

File: packages/airplane/internal/api/client.ts (hunk 1 above)
Comment (yolken-airplane) [reply packages/airplane/internal/api/client.ts]: Removed the `agentHost` and `runID` vars. The path ones are used in a few places, so keeping them for now.

**Pull request merged**